HACKING.md

```diff
@@ -42,73 +55,151 @@ Add `$GOPATH/bin` to your `PATH`, so you can run the go programs you install:
-flags are available using
+```
+cd ~/
+git clone https://github.com/<user>/snapd.git
+cd snapd
+```
 
-    go help get
+This process is outlined in the Github documentation [here](https://docs.github.com/en/github/collaborating-with-pull-requests)
 
-At this point you will have the git local repository of the `snapd` source at
-`$GOPATH/src/github.com/snapcore/snapd`. The source for any
-dependent packages will also be available inside `$GOPATH`.
+We value good tests, so when you fix a bug or add a new feature we highly
+encourage you to add tests.
 
-### Dependencies handling
+For more information on testing, please see the Testing section.
 
-Go dependencies are handled via `go mod`.
+### Build host dependencies
 
-Other dependencies are handled via distribution packages and you should ensure
-that dependencies for your distribution are installed. For example, on Ubuntu,
-run:
+Build dependencies can automatically be resolved using `build-dep`on Ubuntu.
 
-    sudo apt-get build-dep ./
+    cd ~/snapd
+    sudo apt-get build-dep .
 
-### Building
+Package build dependencies for other distributions can be found under the
+`packages\`directory.
 
-To build, once the sources are available and `GOPATH` is set, you can just run
+Go module dependencies are automatically resolved at build time.
 
-    go build -o /tmp/snap github.com/snapcore/snapd/cmd/snap
+### Building (natively)
 
-to get the `snap` binary in /tmp (or without -o to get it in the current
-working directory). Alternatively:
+To build the `snap` commandline client:
 
-    go install github.com/snapcore/snapd/cmd/snap/...
+```
+cd ~/snapd
+mkdir -p /tmp/build
+go build -o /tmp/build/snapd ./cmd/snapd
+```
 
-to have it available in `$GOPATH/bin`
+To build the `snapd` REST API daemon:
 
-Similarly, to build the `snapd` REST API daemon, you can run
+```
+cd ~/snapd
+mkdir -p /tmp/build
+go build -o /tmp/build/snapd ./cmd/snapd
+```
 
-    go build -o /tmp/snapd github.com/snapcore/snapd/cmd/snapd
+To build all the`snapd` Go components:
 
-### Contributing
+```
+cd ~/snapd
+mkdir -p /tmp/build
+go build -o /tmp/build ./...
+```
 
-Contributions are always welcome! Please make sure that you sign the
-Canonical contributor license agreement at
-http://www.ubuntu.com/legal/contributors
+### Cross-compiling (example: ARM v7 target)
 
-Snapd can be found on GitHub, so in order to fork the source and contribute,
-go to https://github.com/snapcore/snapd. Check out [GitHub's help
-pages](https://help.github.com/) to find out how to set up your local branch,
-commit changes and create pull requests.
+Install a suitable cross-compiler for the target architecture.
 
-We value good tests, so when you fix a bug or add a new feature we highly
-encourage you to create a test in `$source_test.go`. See also the section
-about Testing.
+```
+sudo apt-get install gcc-arm-linux-gnueabihf
+```
+
+Verify the default architecture version of your GCC cross compiler.
```

Comment:
nitpick: if we want to be very consistent with naming, we can use "cross-compiler" or "cross compiler" instead of both (personally I don't mind it. Just thought I'd mention it)

Reply:
Thanks. Done.